examples/tags-with-hits/app.tsx

```diff
@@ -0,0 +1,400 @@
+            setQuery('');
+          }
+        },
+        getItems({ query }) {
+          return getAlgoliaFacets({
+            searchClient,
+            queries: [
+              {
+                indexName: 'instant_search',
+                facet: 'categories',
+                params: {
+                  facetQuery: query,
+                  maxFacetHits: 3,
+                  filters: mapToAlgoliaNegativeFilters(
+                    state.context.tagsPlugin.tags,
+                    ['categories']
+                  ),
+                },
+              },
+            ],
+            transformResponse({ facetHits }) {
+              return facetHits[0].map((hit) => ({
+                ...hit,
+                facet: 'categories',
+              }));
+            },
+          });
+        },
+        templates: {
+          header() {
+            return (
+              <Fragment>
+                <span className="aa-SourceHeaderTitle">Categories</span>
+                <div className="aa-SourceHeaderLine" />
+              </Fragment>
+            );
+          },
+          item({ item, components }) {
+            return (
+              <div className="aa-ItemWrapper">
+                <div className="aa-ItemContent">
+                  <div className="aa-ItemContentBody">
+                    <div className="aa-ItemContentTitle">
+                      Filter on{' '}
+                      <components.Highlight hit={item} attribute="label" />
+                    </div>
+                  </div>
+                </div>
+                <div className="aa-ItemActions">
+                  <button
+                    className="aa-ItemActionButton aa-DesktopOnly aa-ActiveOnly"
+                    type="button"
+                    title={`Filter on ${item.label}`}
+                  >
+                    <svg
+                      viewBox="0 0 24 24"
+                      fill="none"
+                      stroke="currentColor"
+                      strokeWidth={2}
+                      strokeLinecap="round"
+                      strokeLinejoin="round"
+                    >
+                      <polygon points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3" />
+                    </svg>
+                  </button>
+                </div>
+              </div>
+            );
+          },
+          noResults() {
+            return 'No categories for this query.';
+          },
+        },
+      },
+      {
+        sourceId: 'products',
+        getItems() {
+          return getAlgoliaResults<ProductHit>({
+            searchClient,
+            queries: [
+              {
+                indexName: 'instant_search',
+                query,
+                params: {
+                  clickAnalytics: true,
+                  attributesToSnippet: ['name:10'],
+                  snippetEllipsisText: '…',
+                  filters: mapToAlgoliaFilters(tagsByFacet),
```

Comment:
as far as I can tell groupBy is only used in here, maybe it makes sense to put the two functions together?

Reply:
Handled in https://github.com/algolia/autocomplete/pull/646#discussion_r695743160.